### PR TITLE
DCMAW-9802: Create initial application infrastructure

### PR DESCRIPTION
**Pull request by sandy-may**
​DCMAW-9802
### What changed
- Updated template.yaml to use KMS output and VPC output from the deployed base stacks
- Added infra tests
- Restructured repository so that only functional code is inside the src folder

### Why did it change
To test that the template can be deployed

### Testing
Credential endpoint returns an error that we define in our lambda code:
image

Same for token endpoint: 
image

## Checklists

- [x] There is a ticket raised for this PR that is present in the branch name
- [ ] No PII data logged. [See guidance here](https://govukverify.atlassian.net/wiki/spaces/DCMAW/pages/3502407722/PII+Logging+Considerations)
- [x] Demo to a BA, TA, and the team.
- [ ] Update [README](./blob/main/README.md) with any new instructions or tasks

### CODE REVIEW

File: backend-api/openApiSpecs/async-public-spec.yaml [hunk 1]
```diff
@@ -46,7 +46,7 @@ paths:
                 success:
                   value:
                     {
-                      "sessionId": "37aae92b-a51e-4f68-b571-8e455fb0ec34",
+                      "sessionId": "37aae92b-a51e-4f68-b571-8e455fb0ec34",`
```
Comment (sandy-may) [inline backend-api/openApiSpecs/async-public-spec.yaml:49]:
```suggestion
                      "sessionId": "37aae92b-a51e-4f68-b571-8e455fb0ec34",
```

**Review by J-son1 — APPROVED** [review]
Looks good!

**Review by jamesamooney — CHANGES_REQUESTED** [review]
Do we need to update the `dependabot.yml` with this dir change?

File: backend-api/infra-tests/application.test.ts [hunk 2]
```diff
@@ -35,68 +35,73 @@ describe("Backend application infrastructure", () => {
...
 
-    test("Rate limiting is set", () => {
-      const expectedBurstLimits = {
-        dev: 10,
-        build: 0,
-        staging: 0,
-        integration: 0,
-        production: 0,
-      };
+      test("Rate and burst limit mappings are correctly", () => {
```
[the remark this answers is not in the document]
Comment (sandy-may) [reply backend-api/infra-tests/application.test.ts]: updated

**Review by J-son1 — APPROVED** [review]

**Review by jamesamooney — APPROVED** [review]

**Pull request merged**